CMakeLists.txt

```diff
@@ -30,13 +45,34 @@ if(NOT CMAKE_C_LINK_PIE_SUPPORTED)
                   "If PIE is enabled by default, test auto discovery might fail.")
 endif()
 
-find_program(SPHINX_BIN sphinx-build REQUIRED)
-find_program(LCOV_BIN lcov REQUIRED)
-find_program(GENHTML_BIN genhtml REQUIRED)
-find_program(CLANG_TIDY_BIN NAMES clang-tidy-18 clang-tidy REQUIRED)
-find_program(CLANG_FORMAT_BIN NAMES clang-format-18 clang-format REQUIRED)
+if(BUILD_DOCS)
+    find_program(SPHINX_BIN sphinx-build REQUIRED)
+    find_program(LCOV_BIN lcov REQUIRED)
+    find_program(GENHTML_BIN genhtml REQUIRED)
+    find_program(CLANG_TIDY_BIN NAMES clang-tidy-18 clang-tidy REQUIRED)
+    find_program(CLANG_FORMAT_BIN NAMES clang-format-18 clang-format REQUIRED)
```

Comment:
Those two binaries are not part of the documentation generation, but the testing logic.